benchmarks/Microsoft.AspNetCore.Routing.Performance/Matchers/MatcherSelectCandidatesSmallEntryCountBenchmark.cs

```diff
@@ -72,43 +68,31 @@ private Matcher SetupMatcher(MatcherBuilder builder)
         }
 
         [Benchmark(Baseline = true)]
-        public async Task Baseline()
+        public unsafe void Baseline()
         {
-            var feature = _feature;
-            await _baseline.MatchAsync(_requests[0], feature);
-            Validate(_requests[0], _endpoints[9], feature.Endpoint);
-        }
-
-        [Benchmark]
-        public async Task Dfa()
-        {
-            var feature = _feature;
-            await _dfa.MatchAsync(_requests[0], feature);
-            Validate(_requests[0], _endpoints[9], feature.Endpoint);
-        }
-
-        [Benchmark]
-        public async Task LegacyRoute()
-        {
-            var feature = _feature;
+            var httpContext = _requests[0];
+            var path = httpContext.Request.Path.Value;
+            var segments = new ReadOnlySpan<PathSegment>(Array.Empty<PathSegment>());
 
-            // This is required to make the legacy router implementation work with dispatcher.
-            _requests[0].Features.Set<IEndpointFeature>(feature);
+            var candidates = _baseline.SelectCandidates(path, segments);
 
-            await _route.MatchAsync(_requests[0], feature);
-            Validate(_requests[0], _endpoints[9], feature.Endpoint);
+            var endpoint = candidates.Candidates[0].Endpoint;
+            Validate(_requests[0], _endpoints[9], endpoint);
         }
 
         [Benchmark]
-        public async Task LegacyTreeRouter()
+        public unsafe void Dfa()
         {
-            var feature = _feature;
+            var httpContext = _requests[0];
+            var path = httpContext.Request.Path.Value;
+            var buffer = stackalloc PathSegment[FastPathTokenizer.DefaultSegmentCount];
+            var count = FastPathTokenizer.Tokenize(path, buffer, FastPathTokenizer.DefaultSegmentCount);
+            var segments = new ReadOnlySpan<PathSegment>((void*)buffer, count);
```

Comment:
Is unsafe code required here? I thought a `stackalloc`ed array implicitly converted to a Span/ReadOnlySpan.

Reply:
You might be right. I may be able to get rid of unsafe in a bunch of places.